extensions/git/src/main.ts

```diff
@@ -157,6 +163,99 @@ export async function activate(context: ExtensionContext): Promise<GitExtension>
 	}
 }
 
+async function setVsCodeAsGitEditor(git: Git, outputChannel: OutputChannel, vscodeExe?: string): Promise<boolean> {
+	vscodeExe = vscodeExe || await findVsCodeExeInPath();
+	if (vscodeExe === undefined) {
+		return false;
+	}
+
+	if (await setGitGlobalConfig(git, 'core.editor', `${vscodeExe} --wait`)) {
+		window.showInformationMessage(localize('successGitEditor', "Visual Studio Code has been successfully set as git editor"));
+		return true;
+	} else {
+		window.showErrorMessage(localize('failGitEditor', "Failed to set Visual Studio Code as git editor. See OUTPUT panel form more details"));
+		outputChannel.show();
+		return false;
+	}
+}
+
+async function setVsCodeAsGitDiffTool(git: Git, outputChannel: OutputChannel, vscodeExe?: string): Promise<boolean> {
+	vscodeExe = vscodeExe || await findVsCodeExeInPath();
+	if (vscodeExe === undefined) {
+		return false;
+	}
+
+	if (await setGitGlobalConfig(git, 'difftool.vscode.cmd', `${vscodeExe} --wait --diff "$LOCAL" "$REMOTE"`)
+		&& await setGitGlobalConfig(git, 'diff.tool', 'vscode')) {
+		window.showInformationMessage(localize('successDiffTool', "Visual Studio Code has been successfully set as diff tool"));
+		return true;
+	} else {
+		window.showErrorMessage(localize('failDiffTool', "Failed to set Visual Studio Code as diff tool. See OUTPUT panel form more details"));
```

Comment:
We usually don't tell the user to `see OUTPUT`, but instead provide an action to open the output panel.